cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -884,37 +884,43 @@ toLedgerUpdate :: forall era ledgerera.
                => Ledger.EraCrypto ledgerera ~ StandardCrypto
                => ShelleyBasedEra era
                -> UpdateProposal
-               -> Ledger.Update ledgerera
+               -> Either [ProtocolParametersConversionError] (Ledger.Update ledgerera)
 toLedgerUpdate era (UpdateProposal ppup epochno) =
-    Ledger.Update (toLedgerProposedPPUpdates era ppup) epochno
+  (`Ledger.Update` epochno) <$> (toLedgerProposedPPUpdates era ppup)
 
 
 toLedgerProposedPPUpdates :: forall era ledgerera.
                              ShelleyLedgerEra era ~ ledgerera
                           => Ledger.EraCrypto ledgerera ~ StandardCrypto
                           => ShelleyBasedEra era
                           -> Map (Hash GenesisKey) ProtocolParametersUpdate
-                          -> Ledger.ProposedPPUpdates ledgerera
+                          -> Either [ProtocolParametersConversionError] (Ledger.ProposedPPUpdates ledgerera)
```

Comment:
It's not important to collect all of the errors at once so we can do something simpler such as:

```
Ledger.ProposedPPUpdates . Map.mapKeysMonotonic (\(GenesisKeyHash kh) -> kh) 
  <$> traverse (toLedgerPParamsUpdate era) m
```

Reply:
Ok. I thought that it might be useful to log multiple conversion errors.